pandas/core/nanops.py

```diff
@@ -692,20 +692,20 @@ def nanmean(
     values, mask, dtype, dtype_max, _ = _get_values(
         values, skipna, fill_value=0, mask=mask
     )
-    dtype_sum = dtype_max
+    # dtype_sum = dtype_max
     dtype_count = np.dtype(np.float64)
 
     # not using needs_i8_conversion because that includes period
-    if dtype.kind in ["m", "M"]:
-        dtype_sum = np.dtype(np.float64)
-    elif is_integer_dtype(dtype):
-        dtype_sum = np.dtype(np.float64)
-    elif is_float_dtype(dtype):
-        dtype_sum = dtype
-        dtype_count = dtype
+    # if dtype.kind in ["m", "M"]:
+    #     dtype_sum = np.dtype(np.float64)
+    # elif is_integer_dtype(dtype):
+    #     dtype_sum = np.dtype(np.float64)
+    # elif is_float_dtype(dtype):
+    #     dtype_sum = dtype
+    #     dtype_count = dtype
 
     count = _get_counts(values.shape, mask, axis, dtype=dtype_count)
-    the_sum = _ensure_numeric(values.sum(axis, dtype=dtype_sum))
+    the_sum = _ensure_numeric(values.sum(axis, dtype=np.float64))
```

Comment:
We don't want to cast in integer situations

Reply:
Oh, Do would like to complete this pull or fix this bug?  I have no idea how to fit it in a right way. Thank you for your reply!